pkg/controller/util/node/controller_utils.go

```diff
@@ -298,3 +325,140 @@ func GetNodeCondition(status *v1.NodeStatus, conditionType v1.NodeConditionType)
+	if err != nil {
+		return nil, fmt.Errorf("Error building Tenant Partition Config to %v: %s", ipAddr, err.Error())
+	}
+
+	client, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("Error building kubernetes clientset to %v: %s", ipAddr, err.Error())
+	}
+
+	return client, nil
+}
+
+func GetTenantPartitionInsecureConfig(tenantServerUrl string) (*rest.Config, error) {
+	template := `
+apiVersion: v1
+kind: Config
+clusters:
+- cluster:
+    server: %v
+  name: tenant-partition-cluster
+contexts:
+- context:
+    cluster: tenant-partition-cluster
+  name: node-controller-context
+current-context: node-controller-context
+`
+
+	content := fmt.Sprintf(template, tenantServerUrl)
+
+	tmpfile, err := ioutil.TempFile("", "kubeconfig")
+	if err != nil {
+		return nil, err
+	}
+	defer os.Remove(tmpfile.Name())
+
+	if err := ioutil.WriteFile(tmpfile.Name(), []byte(content), 0666); err != nil {
+		return nil, err
+	}
+
+	return clientcmd.BuildConfigFromFlags("", tmpfile.Name())
+}
+
+func GetTenantPartitionClients(tenantServers []string) ([]clientset.Interface, error) {
+
+	for _, tenantServer := range tenantServers {
+		if err := validateUrl(tenantServer); err != nil {
+			return nil, err
+		}
+	}
+
+	clients := []clientset.Interface{}
+
+	for _, tenantServer := range tenantServers {
+		tenantServerUrl := strings.TrimSpace(tenantServer)
+		client, err := GetTenantPartitionInsecureClient(tenantServerUrl)
+		if err != nil {
+			return nil, fmt.Errorf("Error in getting client for tenant partition (%v) ： %v", tenantServerUrl, err)
+		}
+
+		clients = append(clients, client)
+	}
+
+	return clients, nil
+}
+
+func getTenantPartitionManagerFromClient(client clientset.Interface, stop <-chan struct{}) *TenantPartitionManager {
+	tpInformer := informers.NewSharedInformerFactory(client, 0)
+	go tpInformer.Core().V1().Pods().Informer().Run(stop)
+	go tpInformer.Apps().V1().DaemonSets().Informer().Run(stop)
+	tpAccessor := &TenantPartitionManager{
+		Client:            client,
+		PodInformer:       tpInformer.Core().V1().Pods(),
+		PodGetter:         tpInformer.Core().V1().Pods().Lister(),
+		DaemonSetInformer: tpInformer.Apps().V1().DaemonSets(),
+		DaemonSetStore:    tpInformer.Apps().V1().DaemonSets().Lister(),
+	}
+
+	return tpAccessor
+}
+
+func GetTenantPartitionManagersFromKubeClients(clients []clientset.Interface, stop <-chan struct{}) ([]*TenantPartitionManager, error) {
+	tpAccessors := []*TenantPartitionManager{}
+
+	for _, client := range clients {
+		tpAccessors = append(tpAccessors, getTenantPartitionManagerFromClient(client, stop))
+	}
+
+	return tpAccessors, nil
+}
+
+func GetTenantPartitionManagersFromServerNames(tenantServers []string, stop <-chan struct{}) ([]*TenantPartitionManager, error) {
+	clients, err := GetTenantPartitionClients(tenantServers)
+	if err != nil {
+		return nil, err
+	}
+
+	return GetTenantPartitionManagersFromKubeClients(clients, stop)
+}
+
+func validateUrl(urlString string) error {
```

Comment:
unit test?

Reply:
this is some volatile code. It will be gone once we move to secure mode. It 

> unit test?

Added as suggested.

Reply:
Added as suggested.